src/modules/traceModelContext.js

```diff
@@ -0,0 +1,99 @@
+import { SessionManager } from './sessionManager.js';
+
+class TraceModelContext {
+
+    #traceModel;
+
+    /**
+     * Get the default trace model or a project trace model.
+     *
+     * @param {SessionManager} sessionManager The session manager, used for connections to the Hansken servers
+     * @param {UUID} projectId The projectId to get the traceModel from, or undefined when accessing the default trace model.
+     */
+    constructor(sessionManager, projectId) {
+        this.sessionManager = sessionManager;
+        this.url = projectId ? `/projects/${projectId}/tracemodel`: `/tracemodel`;
+    }
+
+    /**
+     * Get the trace model.
+     *
+     * @returns The trace model
+     */
+    get = () => {
+        if (!this.#traceModel) {
+            return this.sessionManager.gatekeeper(this.url).then(SessionManager.json).then((traceModel) => {
+                this.#traceModel = traceModel;
+                return traceModel;
+            });
+        }
+        return Promise.resolve(this.#traceModel);
+    };
+
+    /**
+     * Get a trace model property by its dotted notation, as used in the queries.
+     *
+     * @param {string} property A dotted trace model property, e.g. `picture.width` or `email.misc.any`
+     * @returns An property model with description, type and unit
+     */
+    property = (property) => {
+        return this.get().then((model) => {
+            const split = property.split('\.');
+            if (split.length === 1) {
```

Comment:
Can you add a comment what `split.length === 1` actually means? Does this mean we're dealing with intrinsic properties or something else?

Reply:
ok